core/src/main/java/tc/oc/pgm/filters/PlayerCountFilter.java

```diff
@@ -0,0 +1,31 @@
+package tc.oc.pgm.filters;
+
+import com.google.common.collect.Range;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.filter.query.MatchQuery;
+import tc.oc.pgm.api.player.MatchPlayer;
+
+/** Check if the number of players in a certain context is within a defined range. */
+public class PlayerCountFilter extends TypedFilter<MatchQuery> {
+  private final Range<Integer> range;
+  private final Filter countFilter;
+
+  public PlayerCountFilter(Range<Integer> range, Filter countFilter) {
+    this.range = range;
+    this.countFilter = countFilter;
+  }
+
+  @Override
+  public Class<? extends MatchQuery> getQueryType() {
+    return MatchQuery.class;
+  }
+
+  @Override
+  public QueryResponse queryTyped(MatchQuery query) {
+    int count = 0;
+    for (MatchPlayer player : query.getMatch().getPlayers()) {
+      if (this.countFilter.query(player.getQuery()) == QueryResponse.ALLOW) count++;
+    }
+    return QueryResponse.fromBoolean(range.contains(count));
```

Comment:
I'd consider some cases where you can short-circuit.

With both the lower bound, and the upper bound, you can break after `n` players allow, or if `remaining + count < n` (it either won't reach minimum, or it can't overflow the maximum).

Some examples:
 - range min=1, count==1 -> can allow
 - range min=10, 5 players remaining to check, count == 4 -> can disallow
 - range max=10, count==11 -> can disallow
 - range max=10, 5 players remaining to check, count == 4 -> can allow

(keep in mind that you need both min and max range to match of the break conditions, or be unbounded)